src/fides/api/api/v1/endpoints/admin.py

```diff
@@ -27,23 +28,28 @@ class DBActions(str, Enum):
     ],
     status_code=status.HTTP_200_OK,
 )
-async def db_action(action: DBActions) -> Dict:
+async def db_action(action: DBActions, revision: Optional[str] = "head") -> Dict:
     """
     Initiate one of the enumerated DBActions.
     """
-    action_text = "upgrade"
 
-    if action == DBActions.reset:
-        if not CONFIG.dev_mode:
-            raise HTTPException(
-                status_code=status.HTTP_501_NOT_IMPLEMENTED,
-                detail="Resetting the application database outside of dev_mode is not supported.",
-            )
+    if action == DBActions.downgrade:
+        downgrade_db(CONFIG.database.sync_database_uri, revision=revision)  # type: ignore[arg-type]
```

Comment:
If downgrade fails, say the downrev is wrong, or it is a downgrade we cannot complete (there are many downgrades that don't actually work) - there will be an uncaught 500 here

Reply:
nice, new try/catch looking good:

Error: Internal Server Error
--



```json
{
  "detail": "Database downgrade failed: PostgreSQL ENUM type requires a name.. Check server logs for more details"
}
```